### PR TITLE
[webpack] simplification of the project webpack file

### PR DESCRIPTION
**Pull request by ridz1208**
### Brief summary of changes
This is necessary to keep supporting projects adding their own modules to webpack from the `webpack-project.config.js`

this functionality was added in https://github.com/aces/Loris/pull/3006

Everything was simplified here the only requirement is to create a `webpack-project.config.js` with the following content

```
const entry = {
    './project/modules/candidate_parameters/js/index.js': './project/modules/candidate_parameters/jsx/index.js',
    './project/modules/media/js/mediaIndex.js': './project/modules/media/jsx/mediaIndex.js',
};

module.exports = entry;
```

This will simply concatenate the lines to the `webpack.config.js` loris file and all the jsx files will be then transpiled.

Also, added the `project/*` to the ignore list for eslint since project files do not need to be linted.

### Testing
- simply override a module with JSX files
- make changes in the overridden JSX
- without this PR, the changes would never take effect because the JSX files in the overridden module are never compiled
- with this PR and the properly configured  `webpack-project.config.js` file. the changes will be transpiled into JS

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -126,7 +126,7 @@ const config = [{
 // Support project overrides
 if (fs.existsSync('./project/webpack-project.config.js')) {
   const projConfig = require('./project/webpack-project.config.js');
-  config.push(projConfig);
+  config[0].entry = Object.assign(config[0].entry, projConfig);
```
Comment (maltheism) [inline webpack.config.js]: This line of code is slick compared to the previous.

**Comment by driusan** [comment]
I don't see why project javascript files shouldn't be linted. Good code practice is good code practice.

**Comment by ridz1208** [comment]
@driusan we dont enforce phpcs, it's on a private repo, collaboration is usually limited to 1 or 2 people.

yes good code practices are recommended but that's not in loris' control in a way

**Comment by driusan** [comment]
@ridz1208 if you really want to ignore eslint in the project directory, why can't you just add a .eslintignore in your project directory?

**Comment by ridz1208** [comment]
@driusan done

**Comment by johnsaigle** [comment]
Failing Travis dude to 
> 1) CandidateListTestIntegrationTest::testPscidLink
NoSuchElementException: Unable to locate element: {"method":"css selector","selector":"body"}

See #4394.

**Comment by ridz1208** [comment]
@johnsaigle thx

**Comment by ridz1208** [comment]
@maltheism @driusan  please re-review

**Pull request merged**